plugins/filter/hashids.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Andrew Pantuso (@ajpantuso) <ajpantuso@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.errors import (
+    AnsibleError,
+    AnsibleFilterError,
+    AnsibleFilterTypeError,
+)
+
+from ansible.module_utils.common.text.converters import to_native
+
+try:
+    from hashids import Hashids
+    HAS_HASHIDS = True
+except ImportError:
+    HAS_HASHIDS = False
+
+
+def initialize_hashids(salt, alphabet, min_length):
+    if not HAS_HASHIDS:
+        raise AnsibleError("The hashids library must be installed in order to use this plugin")
+
+    unchecked_params = {
+        'salt': salt,
+        'alphabet': alphabet,
+        'min_length': min_length,
+    }
+
+    params = {}
+
+    for k in unchecked_params:
+        if unchecked_params[k]:
+            params[k] = unchecked_params[k]
+
+    try:
+        return Hashids(**params)
+    except TypeError as e:
+        raise AnsibleFilterError(
+            "The provided parameters %s are invalid: %s" % (
+                ', '.join(["%s=%s" % (k, v) for k, v in params.items()]),
+                to_native(e)
+            )
+        )
+
+
+def hashids_encode(nums, salt=None, alphabet=None, min_length=None):
+    """Generates a youtube-like hash from a sequence of Ints
+
+       :nums: Sequence of one or more Ints to hash
+       :salt: String to use as salt when hashing
+       :alphabet: String of 16 or more unique characters to produce a hash
+       :min_length: Minimum length of hash produced
+    """
+
+    hashids = initialize_hashids(salt, alphabet, min_length)
+
+    # Handles the case where a single Int is not encapsulated in a list or tuple.
+    # User convenience seems prefferable to strict typing in this case
+    if isinstance(nums, int):
+        nums = [nums]
+
+    try:
+        hashid = hashids.encode(*nums)
+    except TypeError as e:
+        raise AnsibleFilterTypeError(
+            "Data to encode must by a tuple or list of Ints: %s" % to_native(e)
+        )
+
+    return hashid
+
+
+def hashids_decode(hashid, salt=None, alphabet=None, min_length=None):
+    """Generates a youtube-like hash from a sequence of Ints
+
+       :hashid: Hash string to decode
+       :salt: String to use as salt when hashing
+       :alphabet: String of 16 or more unique characters to produce a hash
+       :min_length: Minimum length of hash produced
+    """
+
+    hashids = initialize_hashids(salt, alphabet, min_length)
```

Comment:
Make add some validation that `hashid` is actually a string? And potentially call `to_native` or `to_text` on it to make sure it's not of the wrong type (I don't know what `hashids` expects).

Reply:
Hashids does that check internally [here](https://github.com/davidaurelio/hashids-python/blob/09f879a1967836f1e399c12b745c24291e3e73ad/hashids.py#L240-L241) and [here](https://github.com/davidaurelio/hashids-python/blob/09f879a1967836f1e399c12b745c24291e3e73ad/hashids.py#L18-L20)
The intention as I understand it is that Hashids returns an empty tuple in all cases except if the input is a string which decodes to some ints and those ints when encoded match the input string.
I didn't want to add preprocessing that could upset that behavior, but I can test more and see what happens with a converter up front.

Will also address the rest of these comments, thanks.